mgmtfn/dockplugin/dockplugin.go

```diff
@@ -74,21 +78,28 @@ func InitDockPlugin() error {
 		s.HandleFunc(dispatchPath, logHandler(dispatchPath, dispatchFunc))
 	}
 
-	driverPath := path.Join(pluginPath, driverName) + ".sock"
-	os.Remove(driverPath)
-	os.MkdirAll(pluginPath, 0700)
+	if useUnixSocket {
+		driverPath := path.Join(pluginPath, driverName) + ".sock"
+		os.Remove(driverPath)
+		os.MkdirAll(pluginPath, 0700)
 
-	go func() {
-		l, err := net.ListenUnix("unix", &net.UnixAddr{Name: driverPath, Net: "unix"})
-		if err != nil {
-			panic(err)
-		}
+		go func() {
+			l, err := net.ListenUnix("unix", &net.UnixAddr{Name: driverPath, Net: "unix"})
+			if err != nil {
+				panic(err)
+			}
+
+			log.Infof("docker plugin listening on %s", driverPath)
+			http.Serve(l, router)
+			l.Close()
+			log.Infof("docker plugin closing %s", driverPath)
+		}()
+	} else {
+		go func() {
```

Comment:
http.ListenAndServe() blocks forever. So, i had to create a go routine for it.
Anyway, I'll remove this block of code since we'll just use units sockets for now.